app/assets/javascripts/services/alerts_center_service.js

```diff
@@ -445,6 +445,34 @@ function alertsCenterService(API, $q, $timeout, $document, $uibModal, $http) {
     return objectType;
   }
 
+  function getUrl(providerType, providerId) {
+    var entityUrl = '';
+    var action = '/';
+    var descriptors = providerType.split('::');
+
+    switch (descriptors.pop()) {
```

Comment:
I don't like this client-side specification because it has to be in sync with the controllers all the time. What do you think about using the `restful_redirect_controller` instead? This way the switch could be completely removed and the redirect decision logic could live in that controller.